platform/darwin/src/MGLSymbolStyleLayer.h

```diff
@@ -444,7 +446,9 @@ MGL_EXPORT
 
 /**
  Name of image in sprite to use for drawing an image background. A string with
- {tokens} replaced, referencing the data property to pull from.
+ `{tokens}` replaced, referencing the data property to pull from. (`{token}`
+ replacement is only supported for literal `iconImageName` values; not for
+ property functions.)
```

Comment:
mapbox/mapbox-gl-js#5964 adds some links to anchors on the style specification page. The iOS/macOS runtime styling codegen script doesn’t resolve these anchor links to an absolute URL, but we can [override](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/darwin/scripts/style-spec-overrides-v8.json) the documentation for that platform to talk about something more platform-appropriate. For now, that’d be:

> for equivalent functionality in expressions, use concatenate strings and key paths together with the `stringByAppendingString:` function.

https://github.com/mapbox/mapbox-gl-native/issues/8074#issuecomment-355383677 proposes a more intuitive `join` function, but I can update the override as part of #10726 when I get around to implementing that.